Framework/MDAlgorithms/test/IntegratePeaksMD2Test.h

```diff
@@ -180,6 +180,128 @@ class IntegratePeaksMD2Test : public CxxTest::TestSuite {
+
+  void test_validationPeak2ValuesSphere() { simpleRun(false, 2, 1, 1, false, false); }             // fail A
+  void test_validationPeak2ValuesCylinder() { simpleRun(false, 2, 1, 1, true, false); }            // fail A
+  void test_validationPeak2ValuesEllipsoid() { simpleRun(false, 2, 3, 3, false, true); }           // fail A
+  void test_validationPeak2ValuesEllipsoidAndCylinder() { simpleRun(false, 2, 1, 1, true, true); } // fail A + C
+
+  void test_validationPeak3ValuesSphere() { simpleRun(false, 3, 1, 1, false, false); }  // fail B
+  void test_validationPeak3ValuesCylinder() { simpleRun(false, 3, 1, 1, true, false); } // fail B
+  void test_validationPeak3ValuesEllipsoid() { simpleRun(true, 3, 3, 3, false, true); }
+  void test_validationPeak3ValuesEllipsoidAndCylinder() { simpleRun(false, 3, 1, 1, true, true); } // fail C
+
+  void test_validationPeak4ValuesSphere() { simpleRun(false, 4, 1, 1, false, false); }             // fail A
+  void test_validationPeak4ValuesCylinder() { simpleRun(false, 4, 1, 1, true, false); }            // fail A
+  void test_validationPeak4ValuesEllipsoid() { simpleRun(false, 4, 3, 3, false, true); }           // fail A
+  void test_validationPeak4ValuesEllipsoidAndCylinder() { simpleRun(false, 4, 1, 1, true, true); } // fail A + C
+
+  /// Background Inner Radius
+  // test_validationBkgInner1ValueSphere() { simpleRun(true, 1,1,1, false, false); }
+  //  would duplicate  test_validationPeak1ValueSphere()
+  // test_validationBkgInner1ValueCylinder() { simpleRun(true, 1,1,1, true, false); }
+  //  would duplicate test_validationPeak1ValueCylinder()
+  void test_validationBkgInner1ValueEllipsoid() { simpleRun(true, 3, 1, 3, false, true); }
+  // test_validationBkgInner1ValueEllipsoidAndCylinder() { simpleRun(false, 1,1,1, true, true); } // fail C
+  //  would duplicate test_validationPeak1ValueEllipsoidAndCylinder()
+
+  void test_validationBkgInner2ValuesSphere() { simpleRun(false, 1, 2, 1, false, false); }             // fail A
+  void test_validationBkgInner2ValuesCylinder() { simpleRun(false, 1, 2, 1, true, false); }            // fail A
+  void test_validationBkgInner2ValuesEllipsoid() { simpleRun(false, 3, 2, 3, false, true); }           // fail A
+  void test_validationBkgInner2ValuesEllipsoidAndCylinder() { simpleRun(false, 1, 2, 1, true, true); } // fail A + C
+
+  void test_validationBkgInner3ValuesSphere() { simpleRun(false, 1, 3, 1, false, false); }  // fail B
+  void test_validationBkgInner3ValuesCylinder() { simpleRun(false, 1, 3, 1, true, false); } // fail B
+  //  test_validationBkgInner3ValuesEllipsoid() { simpleRun(true, 3,3,3, false, true); }
+  // would duplicate test_validationPeak3ValuesEllipsoid()
+  void test_validationBkgInner3ValuesEllipsoidAndCylinder() { simpleRun(false, 1, 3, 1, true, true); } // fail C
+
+  void test_validationBkgInner4ValuesSphere() { simpleRun(false, 1, 4, 1, false, false); }             // fail A
```

Comment:
Is it not sufficient to have just one suite of tests with number of radii not equal to 3 or 1 (i.e. have one suite of tests for 2 radii values and get rid of 4 radii tests)

Reply:
I can imagine a bug where 4 values are accepted as it does have at least 3 values, so I think it is worth keeping.

Reply:
OK happy to keep both nradii=2 and 4 tests then